ztools/x509/json.go

```diff
@@ -221,7 +223,25 @@ func (c *Certificate) MarshalJSON() ([]byte, error) {
 	jc.Subject = c.Subject
 	jc.SubjectDN = c.Subject.String()
 	jc.SubjectKeyInfo.KeyAlgorithm = c.PublicKeyAlgorithm
-	jc.SubjectKeyInfo.SPKIFingerprint = c.SPKIFingerprint
+
+	// Include all subject names, DNS names there are
+	for _, obj := range c.Subject.Names {
+
+		switch name := obj.Value.(type) {
+		case string:
+			// Check that this is actually a url and not something else
+			if _, err := url.Parse(name); err == nil {
+				jc.Names = append(jc.Names, name)
+			}
+		}
+	}
+
+	for _, name := range c.DNSNames {
+
+		if _, err := url.Parse(name); err == nil {
+			jc.Names = append(jc.Names, name)
+		}
+	}
```

Comment:
Actually, as it turns out net/url is the wrong library to validate urls. Switching to govalidate, see here. https://www.socketloop.com/tutorials/golang-how-to-validate-url-the-right-way